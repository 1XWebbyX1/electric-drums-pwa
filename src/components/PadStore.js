import React from 'react'
import asyncComponent from './asyncComponent/async';
import store from '../data/drumPadsData'


class PadStore extends React.Component {
  render() { //dynamic import for drum pads
    const DrumPad = asyncComponent(() =>
    	    import('./DrumPad').then(module => module.default)
    	);

      var padStore = store.map( a => {
      return (
         <DrumPad id={a.key} text={a.key} src={a.audio} keyCode={a.keyCode} color={a.color} display={a.text}/>
      );
    });

    return (
        <div class="pad-store">
           {padStore}
      </div>
    );
  }
}

export default PadStore;
