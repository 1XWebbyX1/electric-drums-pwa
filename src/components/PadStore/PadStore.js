import React from 'react'
import DrumPad from './DrumPad/DrumPad';
import store from '../../data/drumPadsData'


class PadStore extends React.Component {
  render() {
      var padStore = store.map( a =>
         <DrumPad id={a.keyName} text={a.keyName} src={a.audio} keyCode={a.keyCode} display={a.text} updateText={this.props.updateText}/>
    );

    return (
        <div class="pad-store">
           {padStore}
      </div>
    );
  }
}

export default PadStore;
