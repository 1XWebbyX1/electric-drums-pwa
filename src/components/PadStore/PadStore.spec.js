// External Depedencies
import React from 'react';
import { shallow, mount, configure } from 'enzyme';
// Our Dependencies
import { expect } from '../utils/chai';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer";
// Our Component
import PadStore from './PadStore';


configure({ adapter: new Adapter() });

describe('PadStore', () => {
  const updateTextMock = jest.fn();

  it("renders correctly", () => {
     const output = renderer.create(
         <PadStore updateText={updateTextMock}/>
   ).toJSON();
     expect(output).to.matchSnapshot(); //to check DrumPads are rendered correctly from data object
  });

  it('renders 9 Drum Pads', () => {
    let app = mount(
       <PadStore updateText={updateTextMock}/>
     );
    expect(app.find('DrumPad').length).to.equal(9);
    app.unmount();
  });

})
