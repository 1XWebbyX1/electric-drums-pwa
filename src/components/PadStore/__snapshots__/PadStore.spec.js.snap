// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PadStore renders correctly 1`] = `
<div
  class="pad-store"
>
  <div
    className="drum-pad"
    id="div-Q"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <h3
      id="h3-Q"
    >
      Q
    </h3>
    <audio
      className="clip"
      id="Q"
      src="https://s3.amazonaws.com/freecodecamp/drums/RP4_KICK_1.mp3"
    />
  </div>
  <div
    className="drum-pad"
    id="div-W"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <h3
      id="h3-W"
    >
      W
    </h3>
    <audio
      className="clip"
      id="W"
      src="https://s3.amazonaws.com/freecodecamp/drums/Brk_Snr.mp3"
    />
  </div>
  <div
    className="drum-pad"
    id="div-E"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <h3
      id="h3-E"
    >
      E
    </h3>
    <audio
      className="clip"
      id="E"
      src="https://s3.amazonaws.com/freecodecamp/drums/Cev_H2.mp3"
    />
  </div>
  <div
    className="drum-pad"
    id="div-A"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <h3
      id="h3-A"
    >
      A
    </h3>
    <audio
      className="clip"
      id="A"
      src="https://s3.amazonaws.com/freecodecamp/drums/Bld_H1.mp3"
    />
  </div>
  <div
    className="drum-pad"
    id="div-S"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <h3
      id="h3-S"
    >
      S
    </h3>
    <audio
      className="clip"
      id="S"
      src="https://s3.amazonaws.com/freecodecamp/drums/Heater-3.mp3"
    />
  </div>
  <div
    className="drum-pad"
    id="div-D"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <h3
      id="h3-D"
    >
      D
    </h3>
    <audio
      className="clip"
      id="D"
      src="https://s3.amazonaws.com/freecodecamp/drums/side_stick_1.mp3"
    />
  </div>
  <div
    className="drum-pad"
    id="div-Z"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <h3
      id="h3-Z"
    >
      Z
    </h3>
    <audio
      className="clip"
      id="Z"
      src="https://s3.amazonaws.com/freecodecamp/drums/Heater-4_1.mp3"
    />
  </div>
  <div
    className="drum-pad"
    id="div-X"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <h3
      id="h3-X"
    >
      X
    </h3>
    <audio
      className="clip"
      id="X"
      src="https://s3.amazonaws.com/freecodecamp/drums/Heater-2.mp3"
    />
  </div>
  <div
    className="drum-pad"
    id="div-C"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <h3
      id="h3-C"
    >
      C
    </h3>
    <audio
      className="clip"
      id="C"
      src="https://s3.amazonaws.com/freecodecamp/drums/Heater-1.mp3"
    />
  </div>
</div>
`;
